draft-ietf-tls-ticketrequests.md

```diff
@@ -124,32 +146,73 @@ Clients MAY send this extension in ClientHello. It contains the following struct
 
 ~~~
 struct {
-    uint8 count;
-} TicketRequestContents;
-~~~
+    uint8 new_session_count;
+    uint8 resumption_count;
+} ClientTicketRequest;
 
-count
-: The number of tickets desired by the client.
+struct {
+    uint8 expected_count;
+} ServerTicketRequest;
 
-Clients can use TicketRequestContents.count to indicate the number of tickets
-they would prefer to receive. Servers SHOULD NOT send more tickets than
-TicketRequestContents.count, as clients will most likely discard any additional
-tickets. Servers SHOULD additionally place a limit on the number of tickets
-they are willing to send, to save resources. Therefore, the number of
-NewSessionTicket messages sent will typically be the minimum of the server's self-imposed
-limit and TicketRequestContents.count.
+struct {
+    select (Handshake.msg_type) {
+        case client_hello: ClientTicketRequest;
+        case encrypted_extensions: ServerTicketRequest;
+    }
+} TicketRequestContents;
+~~~
 
-A server that supports ticket requests MAY echo the "ticket_request" extension in the
-EncryptedExtensions message. If present, it contains a TicketRequestContents structure, where
-TicketRequestContents.count indicates the number of tickets the server expects to send to the client.
+new_session_count
```

Comment:
Can the field explanations be grouped with their struct definitions?

Reply:
Sure!